numpy/_core/src/umath/string_buffer.cpp

```diff
@@ -0,0 +1,30 @@
+/* Ufunc implementations for the StringDType class */
```

Comment:
Do we need this file? Couldn't we just implement `adjust_offsets` in `string_buffer.h`?

Reply:
I thought I needed this to get it to link correctly, but if I guard the implementation with an `extern "C"` block, it seems to link fine. Removed this new file, thanks for poking me on this.

Reply:
Argh, actually it doesn't work. You get errors like this:

```
duplicate symbol '_adjust_offsets' in:
    numpy/_core/_multiarray_umath.cpython-311-darwin.so.p/src_umath_string_ufuncs.cpp.o
    numpy/_core/_multiarray_umath.cpython-311-darwin.so.p/src_umath_stringdtype_ufuncs.cpp.o
ld: 1 duplicate symbol for architecture arm64
```

Is there another way to do this without adding a new `.cpp` file? I could give it a totally unnecessary template parameter? :wink:

Reply:
Inlining the function should probably fix this.

Reply:
Good call, that fixes it, thank you!